lib/Doctrine/ORM/EntityRepository.php

```diff
@@ -221,21 +237,28 @@ public function __call($method, $arguments)
                 $method = 'findOneBy';
                 break;
 
+            case (0 === strpos($method, 'countBy')):
```

Comment:
Could you swap this logic to return early, by moving the existing logic into a `private` method?